common/djangoapps/nodebb/signals/handlers.py

```diff
@@ -155,3 +157,116 @@ def join_group_on_nodebb(sender, event=None, user=None, **kwargs):  # pylint: di
             )
         else:
             log.info('Success: User have joined the group %s successfully' % course.display_name)
+
+
+@receiver(post_save, sender=CourseTeam, dispatch_uid="nodebb.signals.handlers.create_update_groupchat_on_nodebb")
+def create_update_groupchat_on_nodebb(sender, instance, created, **kwargs):
+    """
+    Create group on NodeBB whenever a new team is created
+    OR
+    Update a group whenever existing team changes
+    """
+    team_group_chat = TeamGroupChat.objects.filter(team_id=instance.id).first()
+
+    if created or not team_group_chat:
+        group_info = _get_group_data(instance)
+        status_code, response_body = NodeBBClient().groups.create(**group_info)
+
+        if status_code != 200:
+            log.error(
+                "Error: Can't create nodebb group for the given course %s due to %s" % (
+                    instance.course_id, response_body
+                )
+            )
+        else:
+            TeamGroupChat.objects.create(team_id=instance.id, room_id=response_body['room'])
+            log.info("Successfully created group for course %s" % instance.course_id)
+    else:
+        room_id = team_group_chat.room_id
+        group_info = _get_group_data(instance, is_created=False)
+        status_code, response_body = NodeBBClient().groups.update(room_id=room_id, **group_info)
+
+        if status_code != 200:
+            log.error(
+                "Error: Can't update nodebb group for the given course %s due to %s" % (
+                    instance.course_id, response_body
+                )
+            )
+        else:
+            log.info("Successfully updated group for course %s" % instance.course_id)
+
+def _get_group_data(instance, is_created=True):
+    group_info = {
+        'team': [],
```

Comment:
if team key is going to contain a list, it's name should be teams otherwise it should't be initialised as list. You can ignore this change if nodebb API required data in this format.

Reply:
NodeBB API needs to be changed